pkg/handlers/clusterlogging/handler.go

```diff
@@ -37,38 +40,66 @@ func NewHandlers(opts *config.Options) []handlers.RequestHandler {
 }
 
 func (ext *handler) Process(req *http.Request, context *handlers.RequestContext) (*http.Request, error) {
-	name := context.UserName
-	if context.IsWhiteListed(name) || ext.hasInfraRole(context) {
-		log.Debugf("Skipping additional processing, %s is whitelisted or has the infra role", name)
-		return req, nil
+
+	log.Infof("Received context: UserName: %v", context.UserName)
+	log.Infof("Received context: Roles: %v", context.Roles)
+
+	// if we are getting a .kibana/config url...
+	if ext.isKibanaIndex(*req.URL) {
+		kibanaVersion := ext.getKibanaVersion(*req.URL)
+		if kibanaVersion != "" {
+			log.Infof("Received request: kibana ver: %v", kibanaVersion)
+		}
+
+		log.Infof("Received request: Header: %v", req.Header)
+
+		// use ES client and set headers for Index request (WithHeaders) from req...
+		// we let the security plugin correctly choose the tenant -- so we only seed as required
+		// only seed infra and audit if they are an admin role though...
+
+		// to create index pattern:
+		//IndexWithHeader(".kibana", "doc", "index-pattern:app", ext.getIndexPatternBody("app"), req.Header)
+
+		// to set default index pattern:
+		// note: only do this if it doesn't yet exist...
+		//IndexWithHeader(".kibana", "doc", fmt.Sprintf("config:%s", kibanaVersion), ext.getIndexPatternBody("app"), req.Header)
 	}
-	modRequest := req
-	userInfo := newUserInfo(context)
-	// modify kibana request
-	// seed kibana dashboards
-	ext.documentManager.SyncACL(userInfo)
```

Comment:
Separately... it pains me to say it because of the work I put in, but should we add a PR to remove document manager?

Reply:
yes, likely we should just remove it in a follow up.